replica/tablets.cc

```diff
@@ -361,15 +364,13 @@ class tablet_sstable_set : public sstables::sstable_set_impl {
         : _schema(std::move(s))
         , _tablet_map(tmap.tablet_count())
     {
-        _sstable_sets.reserve(sgm.storage_groups().size());
-        for (const auto& sg : sgm.storage_groups()) {
+        for (const auto& [id, sg] : sgm.storage_groups()) {
             if (sg) {
```

Comment:
else?  can this really happen or is it an internal error with the flat_hash_map?

Reply:
It can go away later. I don't think we need unique_ptr anymore. sg will never be nullptr here.

Reply:
possibly so, when splitting we create new storage_groups anyhow.